recipes/litestar/meta.yaml

```diff
@@ -0,0 +1,536 @@
+{% set version = "2.5.4" %}
+
+package:
+  name: litestar-split
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/l/litestar/litestar-{{ version }}.tar.gz
+  # the SHA256 gets updated by the bot
+  sha256: e76fd2adc53513e20252974fdc169c2b59e82a0d84ab151bd40c22cc04c38b36
+
+build:
+  # the build number gets reset by the bot
+  number: 0
+  noarch: python
+
+requirements:
+  host:
+    - python >=3.8
+  run:
+    - python >=3.8
+
+outputs:
+  - name: litestar
+    build:
+      noarch: python
+      script:
+        - {{ PYTHON }} {{ RECIPE_DIR }}/test_recipe.py --check
+        - {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation --disable-pip-version-check
+      entry_points:
+        - litestar = litestar.__main__:run_cli
+    requirements:
+      host:
+        - pip
+        - hatchling
+        - python >=3.8
+        # for `test_recipe.py` preflight
+        - jinja2
+        - packaging
+        - tomli
+      run:
+        - python >=3.8
+        - anyio >=3
+        - click
+        - exceptiongroup
+        - httpx >=0.22
+        - importlib-metadata
+        - importlib-resources >=5.12.0
+        - msgspec >=0.18.2
+        - multidict >=6.0.2
+        - polyfactory >=2.6.3
+        - pyyaml
+        - rich >=13.0.0
+        - rich-click
+        - typing_extensions
+    test:
+      imports:
+        - litestar
+      commands:
+        - pip check
+      requires:
+        - pip
+    about:
+      home: https://litestar.dev
+      dev_url: https://github.com/litestar-org/litestar
+      doc_url: https://docs.litestar.dev
+      summary: Light-weight and flexible ASGI API Framework
+      license: MIT
+      license_file: LICENSE
+
+  - name: litestar-with-annotated-types
+    build:
+      noarch: python
```

Comment:
I believe that all of these metapackages can be `noarch: generic` instead of `noarch: python` because they are empty packages which do not contain any python modules (they only depend on other packages). Thus, they also do not need a host section or a runtime dependency on python since the litestar base package already constrains the python version.

Please try this, as it will shorten the length of this recipe and probably shorten build times.

Reply:
```
Attempting to finalize metadata for litestar-with-annotated-types
Error: Failed to render jinja template in /home/conda/staged-recipes-copy/recipes/litestar/meta.yaml:
'PYTHON' is undefined
```

Reply:
So the above _could_ perhaps be avoided by splitting into `litestar-feedstock` and a sibling `litestar-extras-feedstock`, both cut from the same version of the upstream sdist. We ended up going this way when splitting `dagster-(libs-)feedstock`.

Reply:
This error is because the recipe is rerendered for every output, but PYTHON is only defined when the current output uses python. So when rerendering for the generic outputs, jinja doesn't know what to do when rendering the part of the for the python parts. However, it doesn't need to know the value of PYTHON becuase that part of the recipe will be discarded anyways.

We can either use Jinja to define PYTHON = 'some-placeholder-so-jinja-does-not-error' when PYTHON is undefined, or we can just use the environment variable syntax in the script. Typically we use `script: {{ PYTHON }}` because it works for both unix and Windows, but here the recipe will be built once on Linux, so we can use `script: $PYTHON`.

Reply:
Sure, even though it's irrelevant whether it builds on windows, I've added the `if PYTHON is not defined` bit, with a note about `noarch: generic`.